tests/deploy/e2e/test_deploy.py

```diff
@@ -921,3 +921,45 @@ async def test_lsm_states(resource_container, server, client, clienthelper, envi
     # One failure, one skip, in the depdencies
     # response is failure
     assert state_map == {rid1: "failed", rid2: "skipped", lsmrid: "failed"}
+
+
+async def test_skipped_for_dependency(resource_container, server, client, clienthelper, environment, agent):
+    """
+    Asserts the state of a resource, on the scheduler, whose dependency has been skipped
+    """
+    version = await clienthelper.get_version()
+
+    resource_container.Provider.set_skip("agent1", "key", 1)
+
+    rid1 = "test::Resource[agent1,key=key]"
+    rid2 = "test::Resource[agent1,key=key2]"
+
+    resources = [
+        {
+            "key": "key",
+            "value": "value",
+            "id": f"{rid1},v={version}",
+            "requires": [],
+            "purged": False,
+            "send_event": True,
+            "receive_events": False,
+        },
+        {
+            "key": "key2",
+            "value": "value",
+            "id": f"{rid2},v={version}",
+            "requires": [rid1],
+            "purged": False,
+            "send_event": True,
+            "receive_events": False,
+        },
+    ]
+    await clienthelper.set_auto_deploy(True)
+    await clienthelper.put_version_simple(resources, version, wait_for_released=True)
+    await clienthelper.wait_for_deployed()
+    scheduler = agent.scheduler
+    assert scheduler._state.resource_state[rid2] == ResourceState(
```

Comment:
Can you also assert rid1 state to show the difference? i.e. I expect non compliant, skipped, not blocked.